Toolset/palettes/script editor/behaviors/revseeditorbehavior.livecodescript

```diff
@@ -2919,3 +2984,1206 @@ private command addStack pStack, pObeyDontSearch, pMarked, pIgnorePasswordProtec
+               if the number of tokens of tNextNonEmptyLine > 0 then exit repeat
+               add 1 to tSubCounter
+            end repeat
+            if token 1 of tNextNonEmptyLine = "case" then
+               next repeat
+            else
+               put tCounter & cr after tCollect
+            end if
+         end if
+         -- end case
+         
+         -- all other foldables
+         put tCounter & cr after tCollect
+      end if -- token 1 of aLine is among the items of tAllowedLineStarts
+   end repeat
+   
+   split tCollect by return as set
+   
+   repeat for each key aLine in pFoldA
+      if tCollect[aLine] then next repeat
+      put "" into pFoldA[aLine]["runs"][1]["text"]
+      put "" into pScriptA[aLine]["metadata"]
+   end repeat
+   
+   repeat for each key aLine in tCollect
+      put "foldable" into item 1 of pScriptA[aLine]["metadata"]
+      if item 2 of pScriptA[aLine]["metadata"] = "folded" then
+         put tFoldingClosed into pFoldA[aLine]["runs"][1]["text"]
+      else
+         put tFoldingOpen into pFoldA[aLine]["runs"][1]["text"]
+      end if
+   end repeat
+   
+   unlock screen
+end findFoldables
+
+on parseStructs pStartLine
+   local tStartLine, tLineText
+   local tHandlerName
+   local tText, tTextA, tTextAKeys
+   local tCurrLine, tCurrFirstToken, tCurrLastToken
+   local tEndStructure
+   local tLineCounter
+   local tEndList
+   
+   put "if,repeat,switch,try" into tEndList
+   
+   local tPendingIf
+   put false into tPendingIf
+   
+   if pStartLine = empty then
+      put word 2 of the clickLine into tStartLine
+   else
+      put pStartLine into tStartLine
+   end if
+   put line tStartLine of field "Script" of me into tLineText
+   
+   ## to discern when search starts at single "else" set a flag
+   local tStartAtSingleElse
+   if token 1 of tLineText = "else" and token - 1 of tLineText = "else" then
+      put true into tStartAtSingleElse
+   end if
+   
+   -- handlers
+   if token 1 of tLineText is among the items of   "on,private,function,command,setProp,getProp,after,before" then
+      if token 1 of tLineText is "private" then
+         put token 3 of tLineText into tHandlername
+      else
+         put token 2 of tLineText into tHandlerName
+      end if
+      
+      put getTokenArray(tStartLine) into tTextA
+      put the keys of tTextA into tTextAKeys
+      sort tTextAKeys numeric ascending
+      
+      repeat for each line tLineCounter in tTextAKeys
+         put tTextA[tLineCounter] into tCurrLine
+         if tCurrLine = "" then next repeat
+         
+         if token 1 of tCurrLine is "end" and token 2 of tCurrLine is tHandlername then
+            exit repeat
+         end if
+      end repeat
+      
+      return tStartLine, tLineCounter - 1
+      exit parseStructs
+   end if -- handler
+   
+   -- conditional if
+   if token 1 of tLineText is "if" or (token 1 of tLineText is "else" and token - 1 of tLineText is "else") then
```

Comment:
To fold "else if" clauses, remove the "token -1 etc part. Could just replace with
if token 1 of tLineText is among the items of "if,"else" then

Reply:
This is a deliberate design decision. I am not sure one should fold "else if" because those are usually short and it would fractionate the folding of an "If" struct even more.

But if the consensus is that it should be foldable then of course your change would make that possible.

Reply:
@bwmilby folding of mulitline comments was an early request for folding. I hesitated to implement it because first I wanted to fold actual code. Then the problem for multi-line "/* */" comment is that there are no clear rules, people nest them which is actully not recommended but does compile. Then the positioning can be anywhere not only at the beginning of a line.

Because of that I decided to put that on hold and do that maybe on a next version if there is a considerable demand for it.

Reply:
@BerndN I’m thinking mostly about the possibly large block comments before each handler (documentation).  I think that it would be beneficial to include these- just stipulate that it only folds this type initially (where the start/end occur at the beginning of a line - first token)

Reply:
I agree about the usefulness of block comments. These can get rather long.
...and nesting block comments isn't (or at least shouldn't be) allowed. Attempts to nest block comments in a handler properly result in a compiler error. Nesting them outside a handler for some reason doesn't cause an error (which may be a bug in the parser), but apparently anything outside a handler is treated as a nonexecutable comment.

Reply:
Yes @mwieder is correct - block comments are C-style, so don't nest. In terms of 'what goes on outside of a handler' - then the parser tends to just skip to something it can pass as a variable or handler definition IIRC... Which is why you can do things like:
```
local sVariable

-- on thisHandlerWillBeIgnored
...
end thisHandlerWillBeIgnored

on thisHandlerWontBeIgnored
end thisHandlerWontBeIgnored
```
I suggested 'fixing' that once - but there were quite a few people who sent evils via the power of email so it is as it has always been!

Reply:
Outside of handlers does kind of make sense for nesting block comments.  That way you can use a block comment to remove an entire handler even if it contained block comments.

Reply:
Regarding folding "else if": what I did in the glx2 editor is
"if" statements fold to the corresponding "end if" for clarity.
"else" and "else if" statements fold to the next "else..." clause or to the "end if" if there are no more.
Not saying this is The Way It Should Be, just that that was my design decision treating "else" and "else if" clauses as equal at the same level, and within the enclosing "if/end if" clause.

Reply:
@bwmilby They don't nest outside of handlers - it is just that things which don't look like handler/var definitions are ignored - whatever they are. I'm pretty sure the handler list parser checks to see if the first token of a line is in the SP_HANDLER token table, and if not just skips to end of line. So in the case of a nested block-comment:
```
/*A fobar /* something
else */B */
on foo
end foo
```
The block comment will be tokenized from A to B which leaves ` */` on the line, which isn't a handler token (on,local,global etc.) and so is just skipped.

Reply:
Right.  What I meant is that you can “nest” a block comment around a single handler and it will execute as if it had nested since the parser will ignore the invalid stuff between where the comment actually ended and the start of the next handler.  Probably better to just comment out the definition statement.

Reply:
@mwieder, @bwmilby
thanks for the comments and suggestions
I will try to implement the suggested additions.
Not sure how to change the current pull-request to add them.
Will find out.

Reply:
Just need to do another commit to your branch.  It will automatically show up on the PR.

Reply:
It will be this evening, but I will take a look.

Reply:
added slash asterisk folding